### DISCUSSION TITLE
General Transaction Status Consultation API

### DISCUSSION THREAD

**Discussion by puneetlakhina** [bacen/pix-api#565] [Q&A]
Please excuse my bad Portugese. Im using Google Translate.

-- Portugese --
* Temos um aplicativo onde comerciantes e compradores conversam entre si.
* Queremos permitir que os comerciantes solicitem um pagamento usando PIX compartilhando sua chave PIX e valor
* Iremos gerar um código PIX estático que inclui a chave. Também podemos adicionar um `txid` gerado automaticamente e incluí-lo no campo do rótulo de referência.

Pergunta: Se usarmos códigos PIX estáticos (gerados a partir da chave e valor do estabelecimento comercial) e incluirmos um `txid` no campo do rótulo de referência, podemos consultar a API: https://bacen.github.io/pix-api/index.html#/Cob/get_cob__txid_ para obter o status da transação?

-- English --
* We have an app where merchants and buyers talk to each other. 
* We want to allow merchants to request a payment using PIX by sharing their PIX key and amount
* We will generate a static PIX code which includes the key. We can also add an auto-generated `txid` and include it in the reference label field. 

Question: If we use static PIX codes (generated using their merchant's key and amount) and include a `txid` in the reference label field, can we consult the API: https://bacen.github.io/pix-api/index.html#/Cob/get_cob__txid_ to get the transaction status?

**Comment by rubenskuhl** [comment]
Yes, you can query the API for static QR-Codes using GET /pix with start date (inicio), end date (fim) and txid. 
If you don't know the specific date for an specific Pix, you can just use Nov-20-2020 as start (the day Pix was launched) and current date/time as end. (preferred time zone is Z for UTC). 

If the query result is a null set, no payment was received. If the query result is a Pix object, it was paid. The Pix object will also tell you how much was paid, since a static QR-Code is accepted regardless of value. Note that there could be more than one Pix in the result set. 

Addendum: you can't use GET /cob endpoint, since this is not a dynamic QR-Code created with POST or PUT /cob. But regardless, webhook will be triggered for the static QR-Code. 

From a design standpoint, I suggested the txid having identifiers for merchant, buyer and an unique number. txid for static QR-Codes can have up to 24 characters, so while not infinite, there is room to have almost unique references.

**Comment by puneetlakhina** [comment]
-- English -- 
Thanks for your response. Am I understanding this correctly then that if we generate a txid ourselves, included it in the reference label field of static code and then consult the PIX API (/pix) offered by **any** of the providers offering PIX API Correctly (#76) , we would get the status of the payment? Even though those providers are not involved in generating the code themselves?

For example our code could look like the following:

`00020126620014br.gov.bcb.pix0140    b76aa9c2-2ec4-4110-954e-ebfe34f05b61520400005303986540512.345802BR5918TESTUSER20588295596006Cidade62160512testtxid12346304648F`

-- Portugese --

Obrigado pela sua resposta. Estou entendendo isso corretamente, então, se nós mesmos gerarmos um txid, incluí-lo no campo do rótulo de referência do código estático e depois consultar a API PIX (/pix) oferecida por **qualquer** dos provedores que oferecem API PIX corretamente (# 76), obteríamos o status do pagamento? Mesmo que esses provedores não estejam envolvidos na geração do código?

Por exemplo, nosso código poderia ser parecido com o seguinte:

`00020126620014br.gov.bcb.pix0140    b76aa9c2-2ec4-4110-954e-ebfe34f05b61520400005303986540512.345802BR5918TESTUSER20588295596006Cidade62160512testtxid12346304648F`

**Reply by rubenskuhl** [reply]
No. Pix API requires credentials from the PSP to provide responses, and PSP is obliged to verify that the QR-Code was issued by the specific account holder. 

While this is a frequent request at this forum (see #62 and #490), Central Bank has been very adamant that this won't happen.

**Reply by puneetlakhina** [reply]
Thank you!

**Reply by puneetlakhina** [reply]
So in essence, any API will only tell us about txids created by the account holder whose credentials are used to make the API call. Correct?

**Comment by rubenskuhl** [comment]
> So in essence, any API will only tell us about txids created by the account holder whose credentials are used to make the API call. Correct?

Exactly.
